core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -506,8 +506,14 @@ sealed abstract class Cause[+E] extends Product with Serializable { self =>
       stackless: Boolean,
       result: List[Unified]
     ): List[Unified] = {
-      def unifyFail(fail: Cause.Fail[E]): Unified =
-        Unified(fail.trace.fiberId, fail.value.getClass.getName(), fail.value.toString(), fail.trace.toJava)
+      def unifyFail(fail: Cause.Fail[E]): List[Unified] = {
+        val unified =
+          Unified(fail.trace.fiberId, fail.value.getClass.getName(), fail.value.toString(), fail.trace.toJava)
+        fail.value match {
```

Comment:
We don't want to violate parametric polymorphism like this.

Reply:
Do we really? I'm not insisting. But when I looked at it, I felt like I, as a ZIO user, would find this behavior helpful -- I would get access to more exceptions, see them in Sentry and so on.
This is a Java interop hack, I wouldn't do things like this in regular code of course :innocent:

Reply:
Yeah we talked about this at the time and I think our experience has been that every time we do we come to regret it.

Reply:
No ugly hack goes unpunished, huh? :smile_cat: